randomstuff/client.py

```diff
@@ -688,69 +702,41 @@ async def get_joke(self, type: str = 'any') -> Joke:
             lang=response.get('lang')
             )
 
-    async def get_safe_joke(self, type: str = 'any') -> Joke:
-        """A highly useful method to get a joke marked safe.
-        
-        Jokes usually returned are safe 90% of the time but this function can filter any 'unsafe' joke.
-        """
-        joke: Joke = await self.get_joke(type)
-        while not joke.safe:
-            joke = await self.get_joke(type)
-        return joke
-
-    async def get_image(self, type:str = 'any') -> str:
-        """
-        This function is a coroutine
+    async def get_image(self, type: str = 'any') -> str:
+        """This function is a coroutine
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
         Equivalent to `Client.get_image`
         """
+        if type == 'any':
+            type = random.choice(IMAGE_TYPES)
+
         if not type in IMAGE_TYPES:
-            raise TypeError(f"Image type not supported. Choose from {IMAGE_TYPES}")
+            raise InvalidType('Invalid image type provided.')
             return
 
-        if type == 'any':
-            type = random.choice(IMAGE_TYPES)
+        _check_coro(self)
 
         if self.version == '3':
             response = await self._session.get(f'{self._base_url}/image/{type}')
 
-        elif self.version == '4':
-            response =  await self._session.get(f'{self._base_url}/image', params={'type': type})
+        elif self.version in ['4', '5']:
+            response = await self._session.get(f'{self._base_url}/image', params={'type': type})
 
         _check_status(response)
-            
-        return (await response.json())[0]
 
-    async def get_waifu(self, plan:str, type:str) -> Waifu:
+        return await response.json()[0]
```

Comment:
This would raise ValueError. You have to parse JSON before iterating it. Do it like:
```py
return (await response.json())[0]
```